### PR TITLE
Add file_name argument to load_state_dict_from_url

### PR DESCRIPTION
**Pull request by MohamedAliRashad**
In hope to add the feature proposed here #39196

### CODE REVIEW

File: torch/hub.py [hunk 1]
```diff
@@ -446,6 +446,7 @@ def load_state_dict_from_url(url, model_dir=None, map_location=None, progress=Tr
     Args:
         url (string): URL of the object to download
         model_dir (string, optional): directory in which to save the object
+        file_name (string, optional): To name the downloaded file and dodge any confusion
```
Comment (ailzhang) [inline torch/hub.py:449]: I'd mention "If `file_name` is not set, the original filename will be kept."

**Review by ailzhang — APPROVED** [review]
Thanks! Please rebase on top of master to make sure CI runs.

File: torch/hub.py (hunk 1 above)
Comment (MohamedAliRashad) [reply torch/hub.py:449]: i think (optional) send the same message

Comment (MohamedAliRashad) [reply torch/hub.py:449]: i think (optional) send the same message

Comment (ailzhang) [reply torch/hub.py:449]: Hmmm I guess that's fine. 
Please rebase on top of master to make sure CI passes, thanks!

**Comment by MohamedAliRashad** [comment]
@ailzhang I think, i wrecked something.
I am sorry but rebasing is still tricky business for me

**Comment by ailzhang** [comment]
@MohamedAliRashad Current diff has a few merge conflicts due to the rebase I think. It might be easier to just redo a new PR from current pytorch master branch. ;)

**Comment by MohamedAliRashad** [comment]
@ailzhang Well Do

**Pull request closed without merging**